front/lib/api/files/upload.ts

```diff
@@ -509,5 +518,77 @@ export async function uploadToCloudStorage(
     });
   }
 
+  // Upsert to the conversation datasource & generate a snippet.
+  // TODO(JIT) the tool output flow do not go through this path.
+  const jitEnabled = await isJITActionsEnabled(auth);
+  if (
+    jitEnabled &&
+    (file.useCase === "conversation" || file.useCase === "tool_output") &&
+    isSupportedPlainTextContentType(file.contentType)
+  ) {
+    // Upsert the file to the conversation datasource.
+    const conversationsSpace =
+      await SpaceResource.fetchWorkspaceConversationsSpace(auth);
+
+    // TODO(JIT) we need to retrieve the conversation linked to the file somehow.
+    const conversationId = 345;
+
+    // Fetch the datasource linked to the conversation...
+    let dataSource = await DataSourceResource.fetchByConversationId(
+      auth,
+      conversationId
+    );
+
+    if (!dataSource) {
+      // ...or create a new one.
+
+      // IMPORTANT: never use the conversation sID in the name or description, as conversation sIDs are used as secrets to share the conversation within the workspace users.
```

Comment:
I thiiiink this is fine as the data source is equally secret right?

Reply:
Indeed, however, we don't need the conversation sid (if we need to retrieve the conversation, it's ID is already an FK)
Also, we never know if for some reason we allow to (or by mistake) iterate on dataSources.